src/functions/interactivity/types.ts

```diff
@@ -22,6 +24,60 @@ export type BlockActionHandler<Definition> = Definition extends
   }
   : never;
 
+export type BlockSuggestionHandler<Definition> = Definition extends
+  FunctionDefinitionArgs<infer I, infer O, infer RI, infer RO> ? {
+    (
+      context: SuggestionContext<FunctionRuntimeParameters<I, RI>>,
+    ): Promise<BlockSuggestionHandlerResponse> | BlockSuggestionHandlerResponse;
+  }
+  : never;
+
+type BlockSuggestionHandlerResponse =
+  | BlockSuggestionHandlerOptionsResponse
+  | BlockSuggestionHandlerOptionGroupsResponse;
+
+type BlockSuggestionHandlerOptionsResponse = {
+  options: MenuOption[];
+};
+
+type BlockSuggestionHandlerOptionGroupsResponse = {
+  option_groups: MenuOptionGroup[];
+};
+
+type MenuOptionGroup = {
+  /**
+   * @description A {@link PlainTextObject} that defines the label shown above this group of options. Maximum length for the `text` property inside this field is 75 characters.
+   */
+  label: PlainTextObject;
+  /**
+   * @description An array of {@link MenuOption} objects that belong to this specific group. Maximum of 100 items.
+   */
+  options: MenuOption[];
+};
+
+type MenuOption = {
+  /**
+   * @description A {@link PlainTextObject} that defines the text shown in the option on the menu. Maximum length for the `text` property inside this field is 75 characters.
+   */
+  text: PlainTextObject;
```

Comment:
This only supports PlainText and doesn't support something like mrkdwn? I think that makes sense, just sanity checking

Reply:
That's correct. See the `text` Field reference in this doc: https://api.slack.com/reference/block-kit/composition-objects#option

> A [text object](https://api.slack.com/reference/block-kit/composition-objects#text) that defines the text shown in the option on the menu. **Overflow, select, and multi-select menus can only use plain_text objects**, while radio buttons and checkboxes can use mrkdwn text objects. Maximum length for the text in this field is 75 characters.